src/java/main/org/apache/zookeeper/ClientCnxnSocketNIO.java

```diff
@@ -49,8 +50,9 @@
 
     private SocketAddress remoteSocketAddress;
 
-    ClientCnxnSocketNIO() throws IOException {
-        super();
+    public ClientCnxnSocketNIO(ZKClientConfig clientConfig) throws IOException {
```

Comment:
I used getConstructor api which was not able to find default or projected constructor
now i am using getDeclaredConstructor API, which works as epected
Fixed it